bytestream.go

```diff
@@ -73,22 +77,46 @@ func ByteStreamConsumer(opts ...byteStreamOpt) Consumer {
 		}
 		b := buf.Bytes()
 
-		if bu, ok := data.(encoding.BinaryUnmarshaler); ok {
-			return bu.UnmarshalBinary(b)
-		}
+		switch dest := data.(type) {
+		case encoding.BinaryUnmarshaler:
+			return dest.UnmarshalBinary(b)
+		case *string:
+			*dest = string(b)
```

Comment:
where is dest used? I don't see a loop or usage outside of the case statement

Reply:
"dest" is a pointer to data. The "*dest" assignment therefore modifies the content of the "data" argument.